src/compute_horde_prompt_solver/run.py

```diff
@@ -91,20 +106,78 @@ def process_file(
         json.dump(responses, f, indent=2)
 
 
+def _run_server(start_server_event, seed_queue, args):
+    if not args.server:
+        return
+
+    start_server_event.wait()
+
+    app = Flask("compute_horde_prompt_solver")
+
+    @app.route("/health")
+    def server_healthcheck():
+        return {"status": "ok"}
+
+    @app.route("/execute-job", methods=["POST"])
+    def execute_job():
+        try:
+            from flask import request
+
+            seed_raw = request.json.get("seed")
+            seed = int(seed_raw)
+            seed_queue.put(seed)
+            return "", 200
+        finally:
+            # The seed_queue.put(seed) can fail (request not having int seed etc.),
+            # so we always put a None to make sure process is terminated when the view returns.
+            seed_queue.put(None)
+
+    app.run(
+        host="0.0.0.0",
+        port=args.server_port,
+        debug=False,
+    )
+
+
 def main():
     args = parse_arguments()
 
-    # Set deterministic behavior
-    set_deterministic(args.seed)
+    start_server_event = mp.Event()
+    seed_queue = mp.Queue()
+    process = mp.Process(target=_run_server, args=(start_server_event, seed_queue, args))
+    process.start()
 
     args.output_dir.mkdir(parents=True, exist_ok=True)
 
+    # TODO: make sure if it is ok to load model before calling set_deterministic()!!
     model = setup_model(args.model)
```

Comment:
I made 2-3 runs with same/different seeds and they seem to be behaving similarly. Sill I am a bit skeptical.